XCHammerAssets/bazel_build_service_setup.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Values set at project generation time, for reference check `BazelExtensions/xcodeproject.bzl` => `_install_xcode_project` rule
+BUILD_SERVICE_BAZEL_EXEC_ROOT=__BAZEL_EXEC_ROOT__
+
+# Check `BazelExtensions/source_output_file_map_aspect.bzl`, xcbuildkit needs to know what pattern to look for to pre-load indexing information
+BUILD_SERVICE_SOURCE_OUTPUT_FILE_MAP_SUFFIX=source_output_file_map.json
+
+# Load DataStore related paths, used below to setup the DataStore or undo the setup if the build service is not installed
+WORKSPACE_ROOT=$(echo "$BUILD_DIR" | sed -e 's/\/Build.*//g')
+
+if [[ ! -d $WORKSPACE_ROOT ]]; then
+    echo "[ERROR] Failed to setup build service. Workspace not found at $WORKSPACE_ROOT"
+    exit 1
+fi
+
+INDEX_DATA_STORE=$WORKSPACE_ROOT/Index/DataStore
+INDEX_DATA_STORE_BACKUP=$WORKSPACE_ROOT/Index/DataStore.default
+
+undoDataStoreSetupIfNecessary () {
```

Comment:
From a structure perspective do you think it is better to run this script from the build service? It seems like we can encapsulate such an implementation detail of indexing there / not a concern a project generator necessarily needs to deal with

Reply:
Mainly, the the bulk of purpose of the project generator would be to write the config for a given project - and the build service would do any of this symlinking

Reply:
Yep, this makes a lot of sense. Moved the DataStore setup logic to `xcbuildkit`. See latest commit: https://github.com/jerrymarino/xcbuildkit/pull/49/commits/1eb4ed9f2026b35cece333308ba9623acede90f3